custom-aa/test/main.test.ts

```diff
@@ -0,0 +1,104 @@
+import { Wallets } from "../../tests/testData";
+
+describe("Custom aa", function () {
+  let result: any;
+  let factory: Contract;
+  let multiSig: any;
+  const helper = new Helper();
+
+  describe("Factory", function () {
+    before(async function () {
+      this.timeout(10000);
+      factory = await deployFactory(localConfig.privateKey);
+    });
+
+    it("Should have a correct address", async function () {
+      result = await helper.isValidEthFormat(factory.address);
+      expect(result).to.be.true;
+    });
+
+    it("Should have a tx hash that starts from 0x", async function () {
+      result = factory.deployTransaction.hash;
+      expect(result).to.contains("0x");
+    });
+
+    it("Should have the confirmations value as 0", async function () {
+      result = factory.deployTransaction.confirmations;
+      expect(result).to.equal(0);
+    });
+
+    it("Should have the From value as a rich wallet address", async function () {
+      result = factory.deployTransaction.from;
+      expect(result).to.equal(Wallets.richWalletAddress);
+    });
+
+    it("Should have the Signer address value as a rich wallet address", async function () {
+      result = factory.signer;
+      expect(result.address).to.equal(Wallets.richWalletAddress);
+    });
+  });
+
+  describe("Multisig", function () {
+    before(async function () {
+      multiSig = await deployMultisig(factory.address);
+    });
+
+    it("Should be deployed and have a address", async function () {
+      result = await helper.isValidEthFormat(multiSig[1]);
+      expect(result).to.be.true;
+    });
+
+    it("Should have a tx hash that starts from 0x", async function () {
+      result = multiSig[0];
+      expect(result).to.contains("0x");
+    });
+
+    it("Should have a balance with the 8000000000000000 value initially", async function () {
+      result = multiSig[2];
+      expect(result).to.equal("8000000000000000");
+    });
+
+    it("Should have a balance with the value 7470397250000000 eventually", async function () {
+      result = multiSig[3];
+      expect(result).to.equal("7470397250000000");
+    });
+
+    it("Should have the Multisig balance before a transaction more than after", async function () {
+      const multiSigBalanceBeforeTx = Number(multiSig[2]);
+      const multiSigBalanceAfterTx = Number(multiSig[3]);
+      expect(multiSigBalanceBeforeTx).to.be.greaterThan(multiSigBalanceAfterTx);
+    });
+
+    it("Should have the Signed tx hash that starts from 0x", async function () {
+      result = multiSig[4];
+      expect(result).to.contains("0x");
+    });
+
+    it("Should have the Multisig nonce as 0 initially", async function () {
+      result = multiSig[6];
+      expect(result).to.equal(0);
+    });
+
+    it("Should have the Multisig nonce as 1 eventualy", async function () {
+      result = multiSig[7];
+      expect(result).to.equal(1);
+    });
+
+    it("Should have the Multisig nonce less than after", async function () {
```

Comment:
I'd also again suggest adding tests that fail, for example trying to send a value larger than the balance. Also some simple account actions, such as sending ETH somewhere and verifying it's been sent as the transaction flow that's part of the deployment process does not send any ETH value.

Reply:
Yes this transaction has the multisig sign a transaction but no ETH is sent on the transaction. 

As for the failing test and the error message: what does the error message read? Can that error not be caught and verified it is thrown if it reads something along the lines of the account not having enough balance to execute the transaction?